backend/question-service/src/controllers/questionController.ts

```diff
@@ -117,7 +123,94 @@ export const updateQuestion = async (
       question: updatedQuestion,
     });
   } catch (error) {
-    console.log(error);
-    res.status(500).json({ message: "Server error", error });
+    res.status(500).json({ message: SERVER_ERROR_MESSAGE, error });
+  }
+};
+
+interface QnListSearchFilterParams {
+  page: string;
+  qnLimit: string;
+  title?: string;
+  complexities?: string | string[];
+  categories?: string | string[];
+}
+
+export const readQuestionsList = async (
+  req: Request<unknown, unknown, unknown, QnListSearchFilterParams>,
+  res: Response,
+): Promise<void> => {
+  try {
+    const { page, qnLimit, title, complexities, categories } = req.query;
+
+    if (!page || !qnLimit) {
+      res.status(400).json({ message: PAGE_LIMIT_REQUIRED_MESSAGE });
+      return;
+    }
+
+    const pageInt = parseInt(page, 10);
+    const qnLimitInt = parseInt(qnLimit, 10);
+
+    if (pageInt < 1 || qnLimitInt < 1) {
+      res.status(400).json({ message: PAGE_LIMIT_INCORRECT_FORMAT_MESSAGE });
+      return;
+    }
+
+    const query: any = {};
+
+    if (title) {
+      query.title = { $regex: new RegExp(title, "i") };
+    }
+
+    if (complexities) {
+      query.complexity = {
+        $in: Array.isArray(complexities) ? complexities : [complexities],
+      };
+    }
+
+    if (categories) {
+      query.category = {
+        $in: Array.isArray(categories) ? categories : [categories],
+      };
+    }
+
+    const filteredTotalQuestions = await Question.countDocuments(query);
+    if (filteredTotalQuestions == 0) {
+      res.status(404).json({ message: QN_NOT_FOUND_MESSAGE });
+      return;
+    }
+    const filteredTotalPages = Math.ceil(filteredTotalQuestions / qnLimitInt);
+
+    const filteredQuestions = await Question.find(query)
+      .skip((pageInt - 1) * qnLimitInt)
+      .limit(qnLimitInt);
+
+    res.status(200).json({
+      message: QN_RETRIEVED_MESSAGE,
+      pages: filteredTotalPages,
```

Comment:
I think in the frontend will need the total number of questions for the pagination to work

![image](https://github.com/user-attachments/assets/ec11647b-a61a-4935-b09e-1d63bb3f6847)
it's something like this if questions per page = 2 and total question count = 3 (slightly different from our original design)

Would you be able to return the filteredTotalQuestions instead of the filteredTotalPages? thank you~

Reply:
okay